python/tests/test_tables.py

```diff
@@ -2336,6 +2344,34 @@ def test_copy(self):
         t1.edges.clear()
         assert t1 != t2
 
+    def test_equivalent(self):
+        pop_configs = [msprime.PopulationConfiguration(5) for _ in range(2)]
+        migration_matrix = [[0, 1], [1, 0]]
+        t1 = msprime.simulate(
+            population_configurations=pop_configs,
+            migration_matrix=migration_matrix,
+            mutation_rate=1,
+            record_migrations=True,
+            random_seed=1,
+        ).dump_tables()
+        t2 = msprime.simulate(
+            population_configurations=pop_configs,
+            migration_matrix=migration_matrix,
+            mutation_rate=1,
+            record_migrations=True,
+            random_seed=1,
+        ).dump_tables()
+
+        t1.provenances.add_row("random stuff")
+        assert not (t1 == t2)
+        assert t1.equivalent(t2)
+        assert not (t1.equivalent(t2, compare_provenance=True))
+        t1.provenances.clear()
+        t2.provenances.clear()
+        assert t1.equivalent(t2)
```

Comment:
Should also check top-level metadata here, like
```
t1.metadata_schema = tskit.MetadataSchema({"codec": "json", "type": "object"})
t1.metadata = {"hello": "world"}
assert not t1.equivalent(t2)
assert not t2.equivalent(t1)
t2.metadata_schema = t1.metadata_schema
assert not t1.equivalent(t2)
assert not t2.equivalent(t1)
t2.metadata = t1.metadata
assert t1.equivalent(t2)
assert t2.equivalent(t1)
```

Reply:
thank you for this! I was putting off figuring out how to work with the top-level metadata.

Reply:
note: edited that to make it correct just now.